app/models/manageiq/providers/ansible_operation_workflow.rb

```diff
@@ -0,0 +1,109 @@
+class ManageIQ::Providers::AnsibleOperationWorkflow < Job
+  def self.create_job(env_vars, extra_vars, playbook_path, timeout: 1.hour, poll_interval: 1.minute)
+    options = {
+      :env_vars      => env_vars,
+      :extra_vars    => extra_vars,
+      :playbook_path => playbook_path,
+      :timeout       => timeout,
+      :poll_interval => poll_interval,
+    }
+
+    super(name, options)
+  end
+
+  def pre_playbook
+    # A step before running the playbook for any optional setup tasks
+    queue_signal(:run_playbook)
+  end
+
+  def run_playbook
+    env_vars, extra_vars, playbook_path = options.values_at(:env_vars, :extra_vars, :playbook_path)
+
+    response = Ansible::Runner.run_async(env_vars, extra_vars, playbook_path)
+    if response.nil?
+      queue_signal(:abort, "Failed to run ansible playbook", "error")
+    else
+      context[:ansible_runner_response] = response.dump
+
+      started_on = Time.now.utc
+      update_attributes!(:context => context, :started_on => started_on)
+      miq_task.update_attributes!(:started_on => started_on)
+
+      queue_signal(:poll_runner)
+    end
+  end
+
+  def poll_runner
+    response = Ansible::Runner::ResponseAsync.load(context[:ansible_runner_response])
+    if response.running?
+      if started_on + options[:timeout] < Time.now.utc
+        response.stop
+
+        queue_signal(:abort, "Playbook has been running longer than timeout", "error")
+      else
+        queue_signal(:poll_runner, :deliver_on => deliver_on)
+      end
+    else
+      result = response.response
+
+      context[:ansible_runner_return_code] = result.return_code
+      context[:ansible_runner_stdout]      = result.parsed_stdout
+
+      set_status("Playbook failed", "error") if result.return_code != 0
```

Comment:
We should probably also log it? Or the thing is, the response object is the only one holding the result (we are deleting the data from the filesystem). So we need to show the failure somewhere.

I would probably start with log error of the whole response and we can see if I can extract, response.error_message and response.traceback/full_error in a generic way